gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -111,38 +111,106 @@ typedef struct _GstTensorFilterProperties
+ * All the information except the supported acclerator is provided statically.
+ * Accelerators can be provided based on static or dynamic check dependent on framework support.
  */
-typedef struct _GstTensorFilterFramework
+typedef struct _GstTensorFilterFrameworkInfo
 {
   char *name; /**< Name of the neural network framework, searchable by FRAMEWORK property */
   int allow_in_place; /**< TRUE(nonzero) if InPlace transfer of input-to-output is allowed. Not supported in main, yet */
   int allocate_in_invoke; /**< TRUE(nonzero) if invoke_NN is going to allocate outputptr by itself and return the address via outputptr. Do not change this value after cap negotiation is complete (or the stream has been started). */
   int run_without_model; /**< TRUE(nonzero) when the neural network framework does not need a model file. Tensor-filter will run invoke_NN without model. */
   int verify_model_path; /**< TRUE(nonzero) when the NNS framework, not the sub-plugin, should verify the path of model files. */
+  accl_hw *hw_list; /**< List of supported hardwares by the framework.  Positive response of this check does not guarantee successful running of model with this accelerator. */
+} GstTensorFilterFrameworkInfo;
+
+/**
+ * @brief Tensor_Filter Subplugin related events
+ *
+ * These are possible set of events that can be supported by the tensor filter subplugin.
+ */
+typedef enum
+{
+  DESTROY_NOTIFY, /** Free the data element allocated in the invoke callback */
+  RELOAD_MODEL,   /** Reloads the subplugin with newely provided model */
+} event_ops;
+
+
+/**
+ * @brief Tensor_Filter Subplugin's model related info gathering operations
+ */
+typedef enum
+{
+  GET_INPUT_INFO,   /**< Gets the input tensor info */
+  GET_OUTPUT_INFO,  /**< Gets the output tensor info */
+  SET_INPUT_INFO,   /**< Sets the provided input tensor info, and get updated output tensor info */
+} model_info_ops;
+
+/**
+ * @brief Tensor_Filter Subplugin definition
+ *
+ * Common callback parameters:
+ * prop Filter properties. Read Only.
+ * private_data Subplugin's private data. Set this (*private_data = XXX) if you want to change filter->private_data.
+ */
+typedef struct _GstTensorFilterFramework
+{
+  uint64_t version; /**< Version of the struct, can be 0x0 (earlier version (_GstTensorFilterFramework_v1)) or 0x2 (newer version (_GstTensorFilterFramework_v2)) */
+
+  int (*open) (const GstTensorFilterProperties * prop, void **private_data);
+  /**< Optional. tensor_filter.c will call this before any of other callbacks and will call once before calling close
+   *
+   * @param[in] prop read-only property values
+   * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, open() allocates memory for private_data.
+   * @return 0 if ok. < 0 if error.
+   */
+
+  void (*close) (const GstTensorFilterProperties * prop, void **private_data);
```

Comment:
Another thing you need to verify is: Do we really need double pointers for private_data for all callbacks? Don't we need it for open/close only?

Reply:
Yes, I updated this for v1.